tests/installation/language_keyboard/switch_keyboard_layout.pm

```diff
@@ -11,6 +11,8 @@ use Test::Assert ':all';
 
 sub run {
     my $language_keyboard = $testapi::distri->get_language_keyboard();
+    # wait for the English (US) to be selected as on slow architectures it takes some time
+    $language_keyboard->wait_for_keyboard_layout_to_be_selected('English (US)');
```

Comment:
Updated.

Reply:
- I think it is more common and we have been calling it like that since a time ago, 'pre-selected' or 'preselected' (do not need to change it, just neapicking ;) ).

- More important, we are adding another test for this test for doing the hack, but the thing is that we already have a hack, I've just tested and adding to our current hack and sleep all works fine:
```
sub enter_keyboard_test {
    my ($self, $test) = @_;
    # text for test layout cannot be set correctly via rest api, so focus
    # is moved to the control with libyui and typing is done with testapi
    sleep 10;
    $self->{tb_keyboard_test}->set('');
    type_string $test;
}
```
Therefore the problem is in the current hack we already need to do, which already have a comment, so following the logic to not change the test as much as possible as well, we might want to concentrate on making the hack more robust. Also the function you created to wait `wait_for_keyboard_layout_to_be_selected` might not be understood by someone else, because the selection is done properly, it is the next step that happen too fast and is indeed and intentional hack.

We are putting the focus on the `tb_keyboard_test` too soon so if we just replace the `sleep` for some modified `is_shown` where we check only the test UI control and not the combo or both or if it is doesn't work we use the wait_until in this method, then we will have concentrated the hack in just one place. Could we try if that would work?

Reply:
Sorry, but seems like you mixing-up the issues. The solution we use in `enter_keyboard_test` with testapi's `type_string` is added because we need here to really type the text simulating keyboard. The issue fixed in this PR is a different one.

It seems to me, like you did not correctly get what is the issue here. Let me clarify this. The issue is that when the test navigates to "Product Selection" Page, the default value for Keyboard Layout is "Arabic" and then after some time it switches to "English (US)". What I'm doing here is waiting until "English (US)" will be selected. With the 10 seconds sleep you just giving a test more time to pre-set the language, but then it becomes that function for "Keyboard Test" field cares about "Keyboard Layout" combobox behavior that breaks single responsibility principle.

Also, IMHO `wait_for_keyboard_layout_to_be_selected` is descriptive by itself and it can be used not only for waiting until the keyboard layout is pre-selected, but also it can be used at any time when the value is changed and we need to be sure, that the correct language is selected. So, it is not a hack, but a workaround that is used widely in a lot of automation frameworks.

Reply:
we are saying the same more or less, arabic (probably?) or any other language will be internally selected when we test the keyboard, but we have already send the switch and at the end succeeded and set to French, the sleep above probe where is the issue. We need more time to test the keyboard. What I meant is that the switch is not lost or something like that, it just switches slowly from arabic to French that in other arch. So instead of having two hacks, we could have one, let's discuss it offline if still is not clear my point.